### PR TITLE
Add credentials to volume snapshot locations.

### LINKED ISSUE #4862: Expand BSL multiple credentials support to VolumeSnapshotLocations
**Describe the problem/challenge you have**
Originally the per-BSL credentials support was intended to be added to VolumeSnapshotLocations as well. Some initial work was already done for this but was reverted when the VSL work was de-scoped from a prior release (the AWS plugin work to support this remained in place, though). This issue is for going though with the enhancement.

While the need for multiple sets of credentials is not as obvious for VSLs, there's still a use case for it, especially for users who area already setting credentials on their BSLs. For those users, they may not be using the default credentials at all, as they've modified their workflow to set credentials on each BSL they create. This will allow them to follow the same approach with VSLs.

**Describe the solution you'd like**
VSLs should allow optional credentials in the same way as BSLs, with the default credentials being used when they're not specified.

**Anything else you would like to add:**
A PR to implement this will be submitted shortly.

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "The project would be better with this feature added"
- :-1: for "This feature will not enhance the project in a meaningful way"

### PR DESCRIPTION
**Pull request by sseago**
This is analogous to the BSL creds work that was done previously, but for VSLs instead.

Originally the per-BSL credentials support was intended to be added to VolumeSnapshotLocations as well. Some initial work was already done for this but was reverted when the VSL work was de-scoped from a prior release (the AWS plugin work to support this remained in place, though). This issue is for going though with the enhancement.

Signed-off-by: Scott Seago <sseago@redhat.com>

# Please add a summary of your change

# Does your change fix a particular issue?

Fixes #4862

# Please indicate you've done the following:

- [x ] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x ] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

**Comment by reasonerjt** [comment]
We'll target this to post-v1.9

**Comment by reasonerjt** [comment]
@sseago I found the comment by Bridget:
https://github.com/vmware-tanzu/velero/issues/4493#issuecomment-1010285577

> For most cloud providers, it's not possible to take snapshots in different accounts so there isn't really a need to support multiple locations with unique credentials. This is different from the BSL case where it's possible to store the serialized contents of a backup in many storage locations.

So will it cause problems for some cloud providers if we allow them to set different credentials for BSL and VSL?

File: pkg/cmd/cli/snapshotlocation/create.go [hunk 1]
```diff
@@ -54,22 +55,27 @@ func NewCreateCommand(f client.Factory, use string) *cobra.Command {
...
-	Name     string
-	Provider string
-	Config   flag.Map
-	Labels   flag.Map
+	Name       string
+	Provider   string
+	Config     flag.Map
+	Labels     flag.Map
+	Credential flag.Map
+	secretName string
```
Comment (blackpiglet) [inline pkg/cmd/cli/snapshotlocation/create.go:63]: Looks like VSL's create and set command's Options are different.
secretName and secretKey are used as internal values to keep Credential's secret name and key.
Could these two parameters also removed in create command?

Comment (sseago) [reply pkg/cmd/cli/snapshotlocation/create.go]: @blackpiglet I've gone through the BSL credentials commits after the initial implementation (some refactoring and validation changes) -- making similar changes here removes these Options fields, as they're not really necessary.

**Comment by sseago** [comment]
I've gone through the BSL credentials commits after the initial implementation (some refactoring and validation changes) -- making similar changes here.

**Review by blackpiglet — DISMISSED** [review]

**Comment by reasonerjt** [comment]
@sseago Thanks, the code looks good to me.
Could you fix the minor issues with the comment and update the doc accordingly to reflect the changes in API and CLI?

**Comment by sseago** [comment]
@reasonerjt PR updated with comment fixes and doc updates.

File: site/content/docs/main/troubleshooting.md [hunk 2]
```diff
@@ -175,9 +175,9 @@ Follow the below troubleshooting steps to confirm that Velero is using the corre
     <Output should be your credentials>
     ```
 
-### Troubleshooting `BackupStorageLocation` credentials
+### Troubleshooting `BackupStorageLocation` and `VolumeSnapshotLocation credentials
```
Comment (blackpiglet) [inline site/content/docs/main/troubleshooting.md:178]: Missed right back-tick quota mark.

Comment (sseago) [reply site/content/docs/main/troubleshooting.md:178]: Good catch. Will fix.

**Review by shubham-pampattiwar — APPROVED** [review]

**Pull request merged**